### ISSUE TITLE
Clickable property requires "button" to be set as accessibilityRole

### ISSUE THREAD

**Issue by kacieb** [facebook/react-native#31033]
## Description
When a component has an `onPress` property, it should be clickable by TalkBack even if it is not set to be a button. This was discovered [here](https://github.com/facebook/react-native/issues/30947#issuecomment-778414062).

This may require native changes on Android. It may also require an API change. From [this comment](https://github.com/facebook/react-native/issues/30947#issuecomment-780242603), it is common on iOS for anything clickable to be called a "button" according to accessibility. This is *not* true on Android, when only things that look and feel like buttons should be labeled a button.

## React Native version:
0.63

## Steps To Reproduce
```
<Text onPress={() => {console.log("hello!")}} />
```

```
<Text accessibilityRole="button" onPress={() => {console.log("hello!")}} />
```

NOTE: More triage is needed to verify what happens on Android/iOS with these two cases.

## Expected Results
Desired behavior on Android - Text node is clickable even if accessibilityRole is not "button"
Desired behavior on iOS - ?

**Comment by blavalla** [comment]
From a technical perspective on the Android side, the onPress prop should set `view.setClickable(true)` if it's not already.  Usually, by calling `view.setOnClickListener()` this will happen automatically, but maybe we're using a touch listener or some other mechanism to detect the click.

**Comment by chrisglein** [comment]
> This may require native changes on Android. It may also require an API change.

> NOTE: More triage is needed to verify what happens on Android/iOS with these two cases.

There's still some ambiguity here. I'm not sure what the right solution would be based on what you've said here. Can you do some background research and turn this into a more concrete proposal?

**Comment by kacieb** [comment]
Claiming this for now so I can update the issue with more research.

**Comment by fabriziobertoglio1987** [comment]
https://github.com/facebook/react-native/blob/36f3bf2f583596a08a59e1f65016aaf4652fc341/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java#L396-L397
